feat(ras-acc): add helper method to identify premium lists

chickenn00dle opened the pull request:
### All Submissions:

* [ ] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Closes https://app.asana.com/0/1206943664367857/1205992141743582/f

Related to https://github.com/Automattic/newspack-plugin/pull/3105 and https://github.com/Automattic/newspack-blocks/pull/1727

This PR adds a helper method to identify whether a given list is associated with a membership plan.

### How to test the changes in this Pull Request:

There should be no changes in functionality. 

- Smoke tests around newsletter lists tied to membership plans.

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

chickenn00dle committed:
feat(ras-acc): add helper method to identify premium lists

chickenn00dle committed:
fix: add unit tests

chickenn00dle commented on tests/mocks/wc-memberships.php:
```diff
@@ -0,0 +1,45 @@
+<?php // phpcs:ignoreFile
```
Added a mock for wc-memberships plugin testing. Not sure if this is the best way to approach this though, so let me know if there is a cleaner approach!

adekbadek commented on includes/plugins/class-woocommerce-memberships.php:
```diff
@@ -388,6 +388,42 @@ private static function get_user_lists_on_deactivation( $user_id, $membership_id
+	 *
+	 * @return bool
+	 */
+	public static function is_membership_list( $list_id ) {
```
I don't find this method name very helpful. Starting with "is membership" suggests that the subject of this method is a membership, while in fact it's a subscription list. I'd suggest renaming to something more descriptive, like `is_subscription_list_tied_to_plan`.

chickenn00dle replied:
Sure thing! I've updated in https://github.com/Automattic/newspack-newsletters/pull/1502/commits/33fae942c8c09f32f60df78e6a49b4976bebd96e

adekbadek commented on tests/plugins/test-woocommerce-memberships.php:
```diff
@@ -0,0 +1,18 @@
+class Woocommerce_Memberships_Test extends \WP_UnitTestCase {
+	public function test_is_membership_list() { // phpcs:ignore Squiz.Commenting.FunctionComment.Missing
+		$this->assertTrue( Woocommerce_Memberships::is_membership_list( 1 ) );
+		$this->assertFalse( Woocommerce_Memberships::is_membership_list( 0 ) );
```
These tests don't make the use of the method clear. After digging into the code, I see that `0` is an id of a non-existing list, so the method returns false. `1` is apparently an id of a list tied to a membership plan, but it's not evident anywhere in the test. 
A test should be able to be read as a component spec – _given these conditions, code does that, etc_. In this test, the subscription list tied to a membership plan should first be set up, then it's id used in the method test.

chickenn00dle replied:
I see what you mean. I had a hard time moving the setup of the membership plan into the test since getting the memberships relies on the global `wc_memberships_get_membership_plans` method and as far as I can tell, there isn't a simple way to mock this from within the test without adding some more complexity to the wc-memberships mock.

That said, I did move things around and move the setup into a static helper method to try and make things more clear in https://github.com/Automattic/newspack-newsletters/pull/1502/commits/33fae942c8c09f32f60df78e6a49b4976bebd96e

Let me know if this is good enough or if you think there is another strategy I should take here.

adekbadek replied:
Much better!

adekbadek reviewed (changes requested):
Glad to see the tests! Left some remarks. IMO mocking is the right way to go, unless the mock becomes too large.

chickenn00dle committed:
fix: update tests and method name

chickenn00dle commented:
Thanks for the review @adekbadek! Sorry for the delay in addressing the feedback. For some reason I missed the notification on this one.

I tried to address your feedback. Let me know if I need to make further changes. I'm a bit rusty when it comes to PHPUnit so am probably missing some functionality that might make the tests simpler.
